ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/ProcessTransactionSubmission.scala

```diff
@@ -486,14 +511,20 @@ private[kvutils] class ProcessTransactionSubmission(
       .get(stateKey)
       .flatMap {
         _.flatMap { value =>
-          for {
-            contractId <- Some(value.getContractKeyState.getContractId)
-              .filter(_.nonEmpty)
-              .map(decodeContractId)
-            contractStateKey = contractIdToStateKey(contractId)
-            contractState <- inputState.get(contractStateKey).flatMap(_.map(_.getContractState))
-            if contractIsActiveAndVisibleToSubmitter(transactionEntry, contractState)
-          } yield contractId
+          // we cannot check whether the contract is active or not, because we might not have loaded it earlier.
```

Comment:
The name of this function does not imply that we would be checking whether the contract is active or not, hence this explanation feels out of context. Is there a better place where this explanation could go?

Reply:
Unfortunately I don't have a better place. I think this is the best place for the time being.

Reply:
OK